server/registry/internal/storage/client.go

```diff
@@ -63,6 +64,11 @@ func NewClient(ctx context.Context, driver, dsn string) (*Client, error) {
 			c.close()
 			return nil, grpcErrorForDBError(ctx, err)
 		}
+		if err := applyConnectionLimits(db); err != nil {
+			c := &Client{db: db}
+			c.close()
```

Comment:
FWIW, I find this to be a strange pattern we follow: Create a Client in order to call close() on it, which simply calls close() on the db we just constructed it on, only to then throw the client away.

Reply:
Good topic -- it looks like it's allowing us to keep the following code in one place:
```
	sqlDB, _ := c.db.DB()
	sqlDB.Close()
```
That could also be done with `Client{db:db}.close()`... or directly using the above code.

I don't have a strong preference. @seaneganx ?